storage/gcloud/aio/storage/storage.py

```diff
@@ -78,32 +71,50 @@ def __init__(self, project, service_file, token=None, session=None):
         resp.raise_for_status()
         return await resp.json()
 
-    async def upload(self, bucket, object_name, file_data, headers=None,
-                     session=None):
-        # pylint: disable=too-many-arguments
-        # https://cloud.google.com/storage/docs/json_api/v1/how-tos/simple-upload
+    async def upload(self, bucket: str, object_name: str,
+                     file_data, content_type: str,
```

Comment:
It looks like the `aiohttp` library is able to [set the content-type automatically](https://github.com/aio-libs/aiohttp/blob/master/aiohttp/client_reqrep.py#L473). Do we need to be providing this value at all?

Reply:
content-type / mime type is now auto-detected by default. Content type is required when initiating resumable upload and hence must be detected by us and not aiohttp. I have implemented the same method as used by aiohttp (https://github.com/aio-libs/aiohttp/blob/ffaef545a99d67c12b2000d07ce43d885a0cbf1a/aiohttp/payload.py#L160). It would be better to use aiohttp to detect mime type instead of replicating their method but this would be ugly since the detection takes place in a setter of the payload class?

Reply:
Hmm, would it make sense to call out to this method in `aiohttp`? For example, something like:
```
content_type = aiohttp.Payload(None, filename=object_name).content_type
```